### PR TITLE
﻿refactor(core): resolve circular dependencies

### PR DESCRIPTION
**Pull request by LukasBoll**
Eliminated circular dependencies in the core module and reorganized package structure.

Ref #2321

### CODE REVIEW

**Review by lucas-koehler — CHANGES_REQUESTED** [review]
Hi @LukasBoll , thanks for the PR :heart: 
The changes already look mostly good to me and no more circular dependency warnings are shown :)
Besides two inline comments, I suggest renaming the folder `stateToPropsMapper` to simply `mappers`: Not all of the methods are mapping state to props but most of them map either state or dispatch to some kind of props or utility data structure.

File: packages/core/src/jsonFormsCore/jsonFormsCore.ts
[the remark this answers is not in the document]
Comment (LukasBoll) [reply packages/core/src/jsonFormsCore/jsonFormsCore.ts:1]: Yes, you are right, we don't need an extra module for this file, as it only fetches data from the store!

**Comment by LukasBoll** [comment]
> Hi @LukasBoll , thanks for the PR ❤️ The changes already look mostly good to me and no more circular dependency warnings are shown :) Besides two inline comments, I suggest renaming the folder `stateToPropsMapper` to simply `mappers`: Not all of the methods are mapping state to props but most of them map either state or dispatch to some kind of props or utility data structure.

Hi @lucas-koehler,

Thank you for the review! I updated everything according to your suggestions!

**Review by lucas-koehler — APPROVED** [review]
Thanks for the updates @LukasBoll ! LGTM now :)

**Pull request merged**